--- diald-1.0/diald.h.orig	2006-06-07 17:03:57.000000000 +0200
+++ diald-1.0/diald.h	2006-06-07 17:06:01.000000000 +0200
@@ -12,6 +12,9 @@
 #include <stdlib.h>
 #include <stddef.h>
 #  include <time.h>
+#ifndef CLK_TCK
+# define  CLK_TCK      sysconf(_SC_CLK_TCK)
+#endif
 #if HAVE_UNISTD_H
 #  include <unistd.h>
 #endif
--- diald-1.0/utils.c.orig	2006-06-07 17:04:24.000000000 +0200
+++ diald-1.0/utils.c	2006-06-07 17:04:33.000000000 +0200
@@ -1,3 +1,4 @@
+#include <time.h>
 #include <config.h>
 
 #include <diald.h>
@@ -9,7 +10,7 @@
 char *cdate(time_t now)
 {
     static char dt[128];
-    int len;
+    size_t len;
 
     len = strftime(dt, 128, "%c %Z", localtime(&now));
     if (len == 128) dt[len] = 0;
